Prepping for ChIB xyz incorporation.

FrickTobias opened the pull request.

FrickTobias committed:
In the middle of creating xyz_ChIB system.

Has a not working xyz_bc_identifier+stats_function but a working ChIB settings which identifies the correct handles.

FrickTobias committed:
The xyz_bc_active version. Does not work.

FrickTobias committed:
matchSequence non-working. Added pseudo-code framework for xyz_ID after H2/H3.

FrickTobias committed:
xyz dict (non-working). Error: Doesn't xyz fasta file. Currently given as absolute path (~/ChIB/bc_list.fa) in settings.

FrickTobias committed:
xyz file reading running (don't know if working though). Don't give path as string in settings.

FrickTobias committed:
Fully integrated xyz dict reader. Also first step in matching to reads working. Yep, this version actually works.

FrickTobias committed:
Working xyz id identifier and handle stats. Commented, non-working form of stats for xyz bc.

FrickTobias committed:
Fully working ChIB settings. Still need to look at when real_h4 is missing (and real_h1 is present, currently not looking for any bc if so).

FrickTobias committed:
Having problem with real data; currently only looking in r1Seq and never in r2Seq, might be the problem. The hamming distance function returns ValueError.

FrickTobias committed:
Functioning for real data, used try-except. Now loops over r1Seq AN D r2Seq. Having problem with finding xyz when real_h4 is missing.

FrickTobias committed:
Backup before separating handle 2 and 3.

FrickTobias committed:
Version for undetermined_170419 in archive. Should be fully working and ready for continuing the downstream analysis.

FrickTobias committed:
Added the fixinert() function again, realized it extracts the geomic data between handles.

FrickTobias committed:
First step in moving from hard coded X, Y, Z strings for ChIB dictionary

FrickTobias committed:
hard coded XYZ removal: xyz_add_stats, general loop for keys in barcode dictionary.

FrickTobias committed:
hard coded XYZ removal: xyz_add_stats, intact read checking

FrickTobias committed:
Error handling: Dictionary creation, aborts when ChIB_barcode_file is not a file and tells user specifically if setting has not been set yet.

FrickTobias committed:
Removed direction from summary report.

FrickTobias committed:
Documentation/Questions to Erik.

FrickTobias commented:
Didn't find any problems, will be requesting for committing just after approved. The debugging is still having some kind of downstream analysis problems after the for each read function but its the same for the HLA pipeline.

FrickTobias committed:
Notes for future work.

elhb commented:
@Majsstenen I think this looks good can you test to run the 8plex sample (SRR5277650) from the publication data just to make sure we don't break any of the old stuff? //EB

FrickTobias committed:
Found bug, only looking at complete reads

FrickTobias committed:
Real handles for ChIB

FrickTobias committed:
Working on handling inserts, currently own approach

FrickTobias committed:
Inserts are now stored properly, however this is temporary solution (minInsertSize=0 if ChIB)

FrickTobias committed:
Stuff working again, use this not stuff from 2 days ago or less time.

FrickTobias committed:
Stuff working again, use this not stuff from 2 days ago or less time.

FrickTobias committed:
Fully scrapping identifydirection and instead wrote identify_ChIB_handles. Works on perfect match.

FrickTobias committed:
New identifydirection is now compatible with fixInsert().

FrickTobias committed:
Minor bug fixes. Now at the point of fully integrated ChIB for real data. Still have to separate H4-H6.

pontushojer committed:
Update xyz_ID_identifier

Changed missmatch handling loop to only allow matches that have a unique hamming  minimum distance.

FrickTobias committed:
H4-H6 separated and seemingly fully working.

FrickTobias committed:
Merge branch 'master' of https://github.com/Majsstenen/DBS_Analysis

FrickTobias committed:
Removed space in between barcode ID and header in the output insert fasta fila accession row

pontushojer committed:
Added H7prim to ChIB sequences

pontushojer committed:
Added class methods identifyChIBHandles (based on indentify_handles and indentifyDirection) and fixChIBInserts (based on fixInserts). Both used for the ChIB pipline. Some updates done to the xyz_ID_identifier to get barcode in X_Y_Z order and general structuring of the code

pontushojer committed:
Added H7 handle sequence and updated sequnce of others. Renames some for inmport.

pontushojer committed:
Changed readpair class methods used for ChIB analysis. Included UMI in insert file headers.

pontushojer committed:
Removed some print expressions

pontushojer committed:
Added abbility to run matchSequence from a chosen startpostion given with the startOfRead option in combination with a tuple whose first element is Flase. Implemented this to had the start position be the end of the previously found handle.

pontushojer committed:
Fixed issue with inserts not being cut out properly and also construct called OK while not conataining all required handles.

pontushojer committed:
Fixed issue where readpair inserts not cut out properly. Changed insert requirements.

pontushojer committed:
Fixed summary to include info intresting for ChIB experiments.Other small fixes.

pontushojer committed:
Small change to allow inserts

pontushojer committed:
Set dbsMatch to True to get inserts
